packages/react-aria-components/src/GridList.tsx

```diff
@@ -80,14 +90,34 @@ interface GridListInnerProps<T extends object> {
 }
 
 function GridListInner<T extends object>({props, collection, gridListRef: ref}: GridListInnerProps<T>) {
-  let {dragAndDropHooks} = props;
+  let {dragAndDropHooks, keyboardNavigationBehavior = 'arrow', layout = 'stack'} = props;
   let state = useListState({
     ...props,
     collection,
     children: undefined
   });
 
-  let {gridProps} = useGridList(props, state, ref);
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let {disabledBehavior, disabledKeys} = state.selectionManager;
+  let {direction} = useLocale();
+  let keyboardDelegate = useMemo(() => (
+    new ListKeyboardDelegate({
+      collection,
+      collator,
+      ref,
+      disabledKeys,
+      disabledBehavior,
+      layout,
+      direction
+    })
+  ), [collection, ref, layout, disabledKeys, disabledBehavior, collator, direction]);
+
+  let {gridProps} = useGridList({
+    ...props,
+    keyboardDelegate,
+    // Only tab navigation is supported in grid layout.
+    keyboardNavigationBehavior: layout === 'grid' ? 'tab' : keyboardNavigationBehavior
```

Comment:
Wondering if `'arrow'` should be allowed in grid layouts so for example:

https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/layout-grids/#ex2_label can be implemented.

Having the ability to implement the following behaviour for arrow key navigations in grid layout lists regardless of orientation (vertical or horizontal), since layout is only relevant for sighted users, seems useful:

- left/right: navigate to next focusable element, either within the row, or in the next/prev row.
- up/down: Move to prev/next row and attempt to land on the corresponding element.